### PR TITLE
[nnc] Temporarily disable gelu in fuser

### PR DESCRIPTION
**Pull request by bertmaher**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58029 [nnc] Enable CPU fusion inside Facebook
* **#58027 [nnc] Temporarily disable gelu in fuser**
* #58028 [nnc] Handle only the first argument of aten::to
* #58026 Enable cat wo conditionals iff cpu
* #57798 [nnc] Fix float->bool conversion on cpu

Sometimes NNC gelu produces wrong results.

Differential Revision: [D28347705](https://our.internmc.facebook.com/intern/diff/D28347705/)

### CODE REVIEW

**Review by eellison — COMMENTED** [review]
The  current ATen Gelu has one formula on CUDA, and then on CPU either dispatches to MKLDNN if the input is contiguous and mkldnn is built, or it uses the same calculation that NNC is using right now. https://github.com/pytorch/pytorch/blob/3a66a1cb99d7e1c1fe7abf8f390e177b9183a436/aten/src/ATen/native/cpu/Activation.cpp#L267 https://github.com/pytorch/pytorch/blob/3a66a1cb99d7e1c1fe7abf8f390e177b9183a436/aten/src/ATen/native/cuda/Activation.cu#L352

What is the  "wrong" results? I don't know how far we should go for exact bit by bit equivalency to match ATen. 

I would prefer not to disable GELU because it is an extremely common activation function

**Comment by ZolotukhinM** [comment]
Got a repro:
```
import torch

torch._C._jit_set_texpr_fuser_enabled(True)
torch._C._jit_set_texpr_parallel_cpu_enabled(True)
torch._C._jit_override_can_fuse_on_cpu(True)
torch._C._debug_set_fusion_group_inlining(False)

def foo(x):
    return torch.erf(x)

f = torch.jit.script(foo)

for _ in range(10):
    f(torch.randn(1))

for dt in [torch.float32, torch.float64]:
    x = torch.randn(1, dtype=dt)
    x[0] = 0.73751747608184814453125
    exp = f(x).item()
    ref = foo(x).item()

    if exp != ref:
        print(dt)
        print("%.60g" % x.item())
        print("%.60g != %.60g" %(exp, ref))
```

It works for float64, but diverges for float32. The output is:
```
torch.float32
0.73751747608184814453125
0.703055083751678466796875 != 0.70305502414703369140625
```

**Comment by ZolotukhinM** [comment]
Actually, i don't know if that counts as a repro. I for some reason assumed that we should be producing a bitwise identical result for this op, which we apparently don't (both for float32 and float64 cases). The accuracy for `erf` seem to be `1e-7` for float32 and `1e-15` for `float64`. What's interesting is that it drops to `1e-6` for `float32` for `gelu`, but stays `1e-15` for `float64`.

**Comment by bertmaher** [comment]
Interesting, I'd expect 1e-7 to be OK, given it's about 1 ulp.  1e-6 seems more questionable, but the model-level errors I'm getting are way bigger than that.

**Comment by bertmaher** [comment]
> Are we using right lowering for `erf` functions? I suspect we might be using double for float or vice versa. Here is how we lower them:

I'm inspecting the llvm bitcode and we're calling what looks like the right Sleef function:
```
%84 = tail call <8 x float> @Sleef_erff8(<8 x float> %83) #2
```

**Comment by bertmaher** [comment]
Okay, after a long long day of debugging, I finally realized that in fact the whole model level difference can be explained by differences between mkldnn gelu and aten gelu; NNC follows aten gelu, which is fine.  I'm surprised that such small differences can add up, but glad that it looks like this diff is unnecessary.
